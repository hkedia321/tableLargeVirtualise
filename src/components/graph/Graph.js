import React from 'react';
import PropTypes from 'prop-types';
import Highcharts from 'highcharts';
import HighchartsReact from 'highcharts-react-official';
import './graph.css';

const graphStaticOptions = {
    chart: {
        type: 'area',
        height: (7 / 16 * 100) + '%',
        margin: 0
    },
    credits: {enabled: false},
    title: false,
    legend: {
        enabled: false
    },
    xAxis: {
        labels: {
            enabled: false
        },
        title: false,
        tickLength: 0,
        lineColor: 'transparent',  
    },
    yAxis: {
        labels: {
            enabled: false
        },
        title: false,
        gridLineWidth: 0
    },
    plotOptions: {
        series: {
            fillColor: '#eaebfe',
            color: '#c8c9ff',
            marker: {
                radius: 2.5
            }
        },
    },
    tooltip: {
        formatter: function() {
          return (
              `
              <span class="graphTooltipValue">Value: ${this.y}</span>
              <br/>
              <span class="graphTooltipDate">Date: ${this.key} </span> 
               `
          )
        },
      },
  }

const Graph = React.memo(function(props) {
    const options = {
        ...graphStaticOptions,
        series: [{
            data: props.data
          }]
    }
    
      return (
        <HighchartsReact
            data-testid="highChartGraph"
            highcharts={Highcharts}
            options={options}
        />
      )
})

Graph.propTypes = {
    data: PropTypes.array.isRequired
}

export default Graph;