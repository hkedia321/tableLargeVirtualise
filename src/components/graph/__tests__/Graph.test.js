import React from "react"
import { render, cleanup } from "@testing-library/react"
import renderer from "react-test-renderer"
import Graph from ".."

jest.mock(
  "highcharts-react-official",
  () =>
    // eslint-disable-next-line react/prop-types
    function DummyHighcharts({ options }) {
      return <div id="MockedHighcharts">{JSON.stringify(options)}</div>
    }
)

afterEach(cleanup)

const graphDataPoints = [
  ["2020-01-01", 120],
  ["2020-01-02", 175],
  ["2020-01-03", 140],
  ["2020-01-04", 240],
  ["2020-01-05", 200],
  ["2020-01-06", 202],
  ["2020-01-07", 103],
  ["2020-01-08", 35],
  ["2020-01-09", 50],
  ["2020-01-10", 130],
]

it("renders graph without crashing", () => {
  render(<Graph data={graphDataPoints} />)
})

it("matches snapshot", () => {
  const tree = renderer.create(<Graph data={graphDataPoints} />).toJSON()
  expect(tree).toMatchSnapshot()
})
