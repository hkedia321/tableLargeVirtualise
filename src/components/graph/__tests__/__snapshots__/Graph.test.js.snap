// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  id="MockedHighcharts"
>
  {"chart":{"type":"area","height":"43.75%","margin":0},"credits":{"enabled":false},"title":false,"legend":{"enabled":false},"xAxis":{"labels":{"enabled":false},"title":false,"tickLength":0,"lineColor":"transparent"},"yAxis":{"labels":{"enabled":false},"title":false,"gridLineWidth":0},"plotOptions":{"series":{"fillColor":"#eaebfe","color":"#c8c9ff","marker":{"radius":2.5}}},"tooltip":{},"series":[{"data":[["2020-01-01",120],["2020-01-02",175],["2020-01-03",140],["2020-01-04",240],["2020-01-05",200],["2020-01-06",202],["2020-01-07",103],["2020-01-08",35],["2020-01-09",50],["2020-01-10",130]]}]}
</div>
`;
